Make the Transaction Page similar to the wallet page Implementation

song50119 opened the pull request:
### Resolves
Issue #94

### Relevant screenshots or logs

 
 
<img width="912" alt="Screen Shot 2020-05-08 at 15 01 47" src="https://user-images.githubusercontent.com/9800579/81385062-eb4d5300-913c-11ea-8aca-af7179c22358.png">

Commit:
Move transactions page to new architecture

Commit:
ui: Improve Transction filter modal

- Add background

- Press Esc will hide the modal

Commit:
ui: Improve Transactions Filters

Show in the left top, and overlay the Tabs

Commit:
ui: Update Transactions Page filters
- Remove filter button
- Remove dialog
- Add filters on the top of page

Commit:
ui: Style filters

- Remove send, restore button

- Change transactions.go -> transactions_page.go

oshorefueled commented on ui/transactions_page.go:
```diff
@@ -0,0 +1,277 @@
+)
+
+const (
+	PageTransactions                            = "txs"
```
there's no need to abbreviate transactions. You can have it as "transactions", that way all pages follow the same convention.

oshorefueled commented on ui/transactions_page.go:
```diff
@@ -0,0 +1,277 @@
+}
+
+func initTxnWidgets(common *pageCommon,
+	transaction *wallet.Transaction, txWidgets *transactionWdg) {
```
there's no need to break to another line.

oshorefueled commented on ui/transactions_page.go:
```diff
@@ -0,0 +1,277 @@
+
+	if transaction.Txn.Direction == dcrlibwallet.TxDirectionSent {
+		txWidgets.direction, _ = decredmaterial.NewIcon(icons.ContentRemove)
+		txWidgets.direction.Color = common.theme.Color.Danger
```
Icons should be initalized once in the addPages method. By initializing in the layout method, it's re-initalized on every frame event.

oshorefueled commented on ui/transactions_page.go:
```diff
@@ -0,0 +1,277 @@
+	txsPageInsetTop    = 15
+	txsPageInsetLeft   = 15
+	txsPageInsetRight  = 15
+	txsPageInsetBottom = 15
```
these can be in the transactionsPage struct. The idea is to have the variables on the pages localized.

Commit:
ui: Update page struct
- Move const value to page struct
- Add contentRemove, toggleRadioButtonUnchecked, actionCheckCircle to page.go
